client/tests/client.test.js

```diff
@@ -0,0 +1,65 @@
+import {shallow, mount, configure} from 'enzyme';
+import {render, screen} from '@testing-library/react'
+import Adapter from 'enzyme-adapter-react-16';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import axios from 'axios';
+
+import App from '../public/src/components/app.jsx'
+import Options from '../public/src/components/options.jsx'
+import Store from '../public/src/components/store.jsx'
+import Details from '../public/src/components/details.jsx'
+
+configure({ adapter: new Adapter() });
+
+describe('test environment should be set up correctly', () => {
+  test('should always pass', () => {
+    expect(true).toBe(true);
+  })
+})
+
+describe('App should have the correct components rendering', () => {
+  let wrapper = shallow(<App />);
+  test('App should have an options component', () => {
+    expect(wrapper.children().find('Options').exists()).toBe(true);
+  })
+  test('App should have an details component', () => {
+    expect(wrapper.children().find('Details').exists()).toBe(true);
+  })
+  test('App should have an store component', () => {
+    expect(wrapper.children().find('Store').exists()).toBe(true);
+  })
+})
+
+describe('Options should contain the correct elements', () => {
```

Comment:
Your describe blocks are pretty descriptive. Are you planning on adding more client tests, or is this the bulk of it?

Reply:
I spent a lot of time trying to test the functions in App.jsx, but had difficulties since I am using hooks and that made it so much more difficult, since it's now stateless and with no class.